### PR TITLE
Allowed editors domain setting

### PR DESCRIPTION
**Pull request by sethalves**
Allowed Editors setting in DomainServer; domain-list wire protocol includes "you can edit" boolean after uuid; entity server checks canEdit flag of node who sends edit requests; interface does not attempt to do edits if DomainServer sent false for "you can edit".

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: d19ab4c36fbcc3eab04c3c0e3fdb781edacba247 available at: https://gist.github.com/ca94b66f774366d661ae<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4239.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4239.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: bad44c540a692f56af832cfc59fe29f73aad59fa available at: https://gist.github.com/d931bac373f2b6645e37<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4239.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4239.dmg

File: libraries/entities/src/EntityScriptingInterface.cpp [hunk 1]
```diff
@@ -64,36 +73,44 @@ EntityItemID EntityScriptingInterface::identifyEntity(EntityItemID entityID) {
...
         }
-        _entityTree->unlock();
     }
     
     return results;
 }
 
 EntityItemID EntityScriptingInterface::editEntity(EntityItemID entityID, const EntityItemProperties& properties) {
+
+    if (! canEdit()) {
```
Comment (ZappoMan) [inline libraries/entities/src/EntityScriptingInterface.cpp]: coding standard no space between ! and canEdit()

File: libraries/entities/src/EntityScriptingInterface.cpp [hunk 2]
```diff
@@ -134,6 +151,9 @@ EntityItemID EntityScriptingInterface::editEntity(EntityItemID entityID, const E
 
 void EntityScriptingInterface::deleteEntity(EntityItemID entityID) {
 
+    if (! canEdit())
```
Comment (ZappoMan) [inline libraries/entities/src/EntityScriptingInterface.cpp]: coding standard no space between ! and canEdit()
and brackets

**Comment by hifi-gustavo** [comment]
Build console output for commit: 46bf6ea5da091bcd134fa7e5ede20ae77ce7e1e5 available at: https://gist.github.com/739dad6ad6c57db37939<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4239.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4239.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 60a4f48985308ff98cc233fa9a7b578a5a43e086 available at: https://gist.github.com/a8d182c9535dd93d3631<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4239.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4239.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6d5d6b9085fbea93cf1be08278db01314b7c8a73 available at: https://gist.github.com/4aaaf4cb11a6a1d2166d<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4239.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4239.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 839b622a05c69926b0bed53dc3edb0e50f110aef available at: https://gist.github.com/a79188c15eeacd5cf436<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4239.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4239.dmg

File: domain-server/resources/describe-settings.json [hunk 3]
```diff
@@ -73,6 +73,20 @@
             "can_set": true
           }
         ]
+      },
+      {
+        "name": "allowed_editors",
+        "type": "table",
+        "label": "Allowed Editors",
+        "help": "List the High Fidelity names for people you want to be able to make changes to this domain.<br/>An empty list means everyone.",
```
Comment (ZappoMan) [inline domain-server/resources/describe-settings.json]: do you want to change this explanation to clarify what it does?

File: libraries/entities/src/EntityTree.h [hunk 4]
```diff
@@ -162,7 +162,7 @@ class EntityTree : public Octree {
 
     void processRemovedEntities(const DeleteEntityOperator& theOperator);
     bool updateEntityWithElement(EntityItem* entity, const EntityItemProperties& properties, 
-            EntityTreeElement* containingElement);
+                                 EntityTreeElement* containingElement, bool AllowLockChange);
```
Comment (ZappoMan) [inline libraries/entities/src/EntityTree.h]: lower case AllowLockChange

Comment (ZappoMan) [inline libraries/entities/src/EntityTree.h]: can you standardize on "allowLockChange" or "canAdjustLocks" -- it seems like these are used interchangeably, and I'm not sure if they are supposed to mean the same thing or do they mean something different.

**Comment by ZappoMan** [comment]
can you standardize on "allowLockChange" or "canAdjustLocks" -- it seems like these are used interchangeably, and I'm not sure if they are supposed to mean the same thing or do they mean something different.

**Comment by sethalves** [comment]
allowLockChange is a bool argument to the updateEntity functions. It is 
often (but not always) set to the value of canAdjustLocks from the Node.

**Comment by ZappoMan** [comment]
ok - I'll merge once the build is complete

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9539292c223987654a34594c17755e2664609c49 available at: https://gist.github.com/44c823aa7ed0fa574deb<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4239.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4239.dmg

**Pull request merged**